xbmc/music/MusicDatabase.cpp

```diff
@@ -4850,6 +4888,50 @@ bool CMusicDatabase::GetDiscsByWhere(CMusicDbUrl& musicUrl,
   return false;
 }
 
+bool CMusicDatabase::GetDiscTitlesJSON(int idAlbum, CFileItem* item, int idSong /* -1 */)
+{
+  try
+  {
+    std::string strSQL;
+    strSQL =
+        PrepareSQL("SELECT DISTINCT strDiscSubtitle FROM song WHERE song.idAlbum = %i", idAlbum);
+    if (idSong > 0)
+      strSQL += PrepareSQL(" AND song.idSong = %i", idSong);
+    CLog::Log(LOGDEBUG, "%s query = %s", __FUNCTION__, strSQL.c_str());
+    if (!m_pDS->query(strSQL))
+      return false;
+    if (m_pDS->num_rows() == 0)
```

Comment:
This bit as it stands is wrong.  Even an empty title returns one row.   Should be `if (m_pDS->fv("strDiscSubtitle").get_asString().empty())`